src/main/java/io/axoniq/axonserver/connector/impl/AxonServerManagedChannel.java

```diff
@@ -367,9 +351,15 @@ private void verifyConnectionStateChange() {
 
     private void scheduleConnectionCheck(long interval) {
         try {
-            executor.schedule(() -> ensureConnected(true), interval, TimeUnit.MILLISECONDS);
+            if (scheduleGate.compareAndSet(false, true)) {
+                executor.schedule(() -> {
+                    scheduleGate.set(false);
```

Comment:
There is the possibility to have ensureConnected invoked concurrently as scheduleGate.set(false) is invoked before ensureConnected()

maybe we can replace with something like:
                    ensureConnected();
                    scheduleGate.set(false);
                    scheduleConnectionCheck(500);

Reply:
This comment has been discussed in a video meeting and deemed to not cause any issues. 
The gate is merely there to ensure `ensureConnected()` is not invoked _to_ often. 
An additional invocation, which could occur due to the current set up of the gate, does not cause any issues.